### PR TITLE
Ignore signature when parsing function names

### LINKED ISSUE #10: Fix parsing of function names containing the whole signature
More details: https://github.com/ethcore/parity/issues/4122#issue-199917015

### PR DESCRIPTION
**Pull request by andrebeat**
Fixes #10.

### CODE REVIEW

File: src/spec/operation.rs [hunk 1]
```diff
@@ -27,7 +27,14 @@ impl<'a> Deserialize<'a> for Operation {
...
 			"constructor" => from_value(v).map(Operation::Constructor),
-			"function" => from_value(v).map(Operation::Function),
+			"function" => from_value(v).map(|mut f: Function| {
+				// This is a workaround to support non-spec compliant function names,
+				// see: https://github.com/paritytech/parity/issues/4122
+				if let Some(i) = f.name.find('(') {
+					f.name.truncate(i);
+				}
+				Operation::Function(f)
+			}),
```
Comment (debris) [inline src/spec/operation.rs]: What about `event` and `fallback`? Shouldn't we also truncate their names?

cc @tomusdrw

Comment (tomusdrw) [inline src/spec/operation.rs]: Yeah, IMHO we should do it for `events` as well. Not sure about fallback though, in this functions seems that it's already parsed.

@andrebeat Could you add some more tests for fallback (`name: ""` and `name: "()"`) and events?

Comment (andrebeat) [inline src/spec/operation.rs]: The `fallback` event is currently ignored (also according to the spec it doesn't have a name). I just added the same changes for `events` and added more tests. 👍

**Comment by debris** [comment]
thanks! :)

**Pull request merged**